docs/otelcol_cr_spec.md

```diff
@@ -0,0 +1,68 @@
+# OpenTelemetryCollector Custom Resource Specification
+
+The below `OpenTelemetryCollector` custom resource contains all the specification that can be configured. 
+
+```
+apiVersion: opentelemetry.io/v1alpha1
```

Comment:
Is this auto-generated? Will we need to update this manually whenever the CRD changes?

Reply:
This is a manually created doc. I didn't find any effective way of generating docs for CRD spec. :/

Reply:
Alright, I'll merge this one as it's an improvement already, but could you please open a new issue to figure out how to generate docs out of the CRDs?

Reply:
Created a new issue on generating docs from CRD spec [here](https://github.com/open-telemetry/opentelemetry-operator/issues/325)